tests/rptest/services/admin.py

```diff
@@ -78,6 +100,189 @@ def ready(node):
     def _url(node, path):
         return f"http://{node.account.hostname}:9644/v1/{path}"
 
+    def _get_configuration(self, node, namespace, topic, partition):
+        r = requests.get(
+            f"http://{node.ip}:9644/v1/partitions/{namespace}/{topic}/{partition}"
+        )
+        if r.status_code != 200:
+            self.redpanda.logger.error(f"status code: {r.status_code}")
+            self.redpanda.logger.error(f"content: {r.content}")
+            return None
+        return r.json()
+
+    """
+    Method iterates through nodes and checks that the configuration of
+    namespace/topic/partition is the same on all nodes and that all
+    the nodes see the same leader.
+
+    When replication is provided the method also checks that the confi-
+    guration has exactly that amout of nodes.
+
+    When the configuration isn't stable the method returns None
+    """
+
+    def _get_stable_configuration(
+            self,
+            nodes: list[RedpandaNode],
+            topic,
+            partition=0,
+            namespace="kafka",
+            replication: Optional[int] = None) -> Optional[PartitionDetails]:
+        last_leader = -1
+        replicas = None
+        status = None
+        for node in nodes:
+            ip = node.ip
+            self.redpanda.logger.debug(
+                f"requesting \"{namespace}/{topic}/{partition}\" details from {ip})"
+            )
+            meta = self._get_configuration(node, namespace, topic, partition)
+            if meta == None:
+                return None
+            if "replicas" not in meta:
+                self.redpanda.logger.debug(f"replicas are missing")
+                return None
+            if "status" not in meta:
+                self.redpanda.logger.debug(f"status is missing")
+                return None
+            if status == None:
+                status = meta["status"]
+                self.redpanda.logger.debug(f"get status:{status}")
+            if status != meta["status"]:
+                self.redpanda.logger.debug(
+                    f"get status:{meta['status']} while already observed:{status} before"
+                )
+                return None
+            if replicas == None:
```

Comment:
This could be condensed quite a bit using set instead of dict, and a list comprehension:
```
read_replicas = set([r['node_id'] for r in meta["replicas"]])
if replicas is None:
    replicas = read_replicas
else if replicas != read_replicas:
    self.redpanda.logger.debug(f"get conflicting replicas:{read_replicas}")
    return None